node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -687,14 +687,19 @@ private void createContainerData(ContainerNodeSpec nodeSpec) {
         ContainerData containerData = ContainerData.createClean(environment,
                                                                 ContainerName.fromHostname(nodeSpec.hostname));
 
+        // ContainerData only works when root, which is the case only for HostAdmin so far -- config nodes are only used under HostAdmin.
         if (nodeSpec.nodeType.equals(NodeType.config.name())) {
             logger.info("Creating files needed by config server");
             new ConfigServerContainerData(environment, nodeSpec.hostname).writeTo(containerData);
         }
 
-        logger.info("Creating files for message of the day and the bash prompt");
-        new MotdContainerData(nodeSpec, environment).writeTo(containerData);
-        new PromptContainerData(environment).writeTo(containerData);
+        // ContainerData only works when root, which is the case only for HostAdmin so far -- only AWS uses HostAdmin now.
+        if (environment.getRegion().startsWith("aws-")) {
```

Comment:
Please add a method on environment instead of matching the free form and human-readable region.

Reply:
Yes, we should do this properly. I just wanted to un-break CD ASAP. 

So, an `Cloud getCloud()`, or something, is probably a good idea.

Reply:
Ah, that was harder than expected ...

Reply:
Checking against AWS is artificial in any case. Could we just check directly whether using the ContainerData would work?

Reply:
`whoami = "root"`

Reply:
Maybe something like `if (Files.isWritable(ContainerData.containerDataPath)) {`?
Add that as a function to ContainerData?

Reply:
Yes, something like that is clearly the right way when writing is optional. I think the right thing here is to throw an exception when we can't allow this to fail, which is the case for the config server usage. 

For the MotD and the prompt, it's fine to just skip the writing if we can't do it. I'll add a fix like this, but I may choose try-catch in case there are other things that will make us fail that we haven't thought about yet :)

Reply:
It would be nice to know in a catch, whether the exception is an error and should be logged as such w/stacktrace, or whether it is fine and can be ignored. Testing for Files.isWriteable is better in that respect.

Reply:
That's why I was thinking to try-catch only the part that is allowed to fail.

Reply:
Ok, my point was that I think it's better to `try-catch`, or check `Files.isWritable`, only for the part where we can continue without success.